src/rules/no-default-export.js

```diff
@@ -15,7 +15,10 @@ module.exports = {
 
   create(context) {
     // ignore non-modules
-    if (context.parserOptions.sourceType !== 'module') {
+    if (
+      context.parserOptions.sourceType !== 'module'
+      && (!context.languageOptions || context.languageOptions.sourceType !== 'module')
```

Comment:
this line isn't covered; how can we cover it?

Reply:
I'm pretty sure it is covered, it's that codecov is getting confused - it'll probably be because it's only ever covered in one direct.

Once you've cherry-picked some stuff and the diff is smaller, I'll try having a closer look to see if I can get codecov sorted out

Reply:
Sounds like a plan!

lack of coverage needn't block the PR, it's just nice to handle.

Reply:
so I'm not too sure what can be done easily about coverage as it doesn't look like there's any special setup to handle the matrixes (i.e. there's no attempt to merge coverage), and my understanding with how flat config works means that this should always be an "either or" path.

At least one thing we could do to try reducing the amount of uncovered lines would be to introduce a new helper function - I was considering doing that but didn't when it turned out there were only three instances of it.

Let me know what you'd prefer to do

Reply:
codecov definitely merges the coverage from every single run, that's why it's the best code coverage tool :-)

Reply:
hmm I don't think that matches our experience over in `eslint-plugin-jest`, but I might be forgetting about some config setting or edge case or something.

While we wait to see what CI says, I'll start by double checking that all three of these do actually seem covered locally in case I've mixed something up

Reply:
ok so `unambiguous` is the only one that doesn't have a test that fails if we remove this:

```
❯ npm run  mocha tests/src/rules/no-default-export.js tests/src/rules/no-named-export.js tests/src/rules/unambiguous.js

> eslint-plugin-import@2.30.0 mocha
> cross-env BABEL_ENV=test nyc mocha tests/src/rules/no-default-export.js tests/src/rules/no-named-export.js tests/src/rules/unambiguous.js



  ․․․․․․․․․․․․․․․․․․․․․․․․․!․․․․․․․․․․․․․․․․․․․․․!!!!․․․․․․․․․․․․

  58 passing (294ms)
  5 failing

  1) no-default-export invalid export default from "foo.js":

      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: Should have 1 error but had 0: []
      + expected - actual

      -0
      +1

      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:998:24)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:1274:33)
      at processImmediate (node:internal/timers:478:21)

  2) no-named-export invalid export { a, b } from "foo.js":

      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: Should have 1 error but had 0: []
      + expected - actual

      -0
      +1

      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:998:24)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:1274:33)
      at processImmediate (node:internal/timers:478:21)

  3) no-named-export invalid export type UserId = number;:

      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: Should have 1 error but had 0: []
      + expected - actual

      -0
      +1

      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:998:24)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:1274:33)
      at processImmediate (node:internal/timers:478:21)

  4) no-named-export invalid export foo from "foo.js":

      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: Should have 1 error but had 0: []
      + expected - actual

      -0
      +1

      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:998:24)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:1274:33)
      at processImmediate (node:internal/timers:478:21)

  5) no-named-export invalid export Memory, { MemoryValue } from './Memory':

      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: Should have 1 error but had 0: []
      + expected - actual

      -0
      +1

      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:998:24)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:1274:33)
      at processImmediate (node:internal/timers:478:21)
```

Reply:
oh right, it's that these lines just don't have coverage already on `main` - gotcha

Reply:
https://github.com/import-js/eslint-plugin-import/pull/2996/commits/e5a76f0183511d0c61c52b1b58bf569c61761f58 - feel free to cherry-pick since it applies to `main` as well 🙂